libs/barista-components/table/src/table.ts

```diff
@@ -370,14 +384,25 @@ export class DtTable<T> extends _DtTableBase<T> implements OnDestroy {
 
   /** @internal Exports the filtered source data from the dataSource. */
   // Note: this is different from the display text, see instead _exportDisplayData().
-  _exportFilteredData(): void {
-    const exportData = this._filteredData;
-    if (this.isEmptyDataSource || typeof exportData[0] != 'object') {
-      return;
+  _exportFilteredData(selectedData?: T[]): void {
+    //nothing to export
+    if (this.isEmptyDataSource) return;
+    //not using DTDataSource, fallback to this._data instead of this._filteredData
+    let exportData: readonly T[];
+    if (selectedData) {
+      exportData = selectedData;
+    } else {
+      exportData =
+        Array.isArray(this._filteredData) &&
+        typeof this._filteredData[0] == 'object'
```

Comment:
We have an `isObject` in the core module, you might be able to use this one as well. 
https://github.com/dynatrace-oss/barista/blob/f81416513137d54e281211046a6abc71350ed1db/libs/barista-components/core/src/util/type-util.ts#L54-L56